packages/core/src/electron-browser/menu/electron-main-menu-factory.ts

```diff
@@ -161,9 +163,12 @@ export class ElectronMainMenuFactory {
 
     protected execute(command: string): void {
         this.commandRegistry.executeCommand(command).catch(() => { /* no-op */ });
-        if (this.commandRegistry.getToggledHandler(command)) {
-            const menu = electron.remote.Menu;
-            menu.setApplicationMenu(menu.getApplicationMenu());
+        if (this.commandRegistry.isVisible(command)) {
+            const menu = electron.remote.Menu.getApplicationMenu();
+            if (menu) {
+                menu.getMenuItemById(command).checked = this.commandRegistry.isToggled(command);
+                electron.remote.Menu.setApplicationMenu(menu);
```

Comment:
Tested electron on your PR just now, this reintroduce the issue where ever window was sharing the same title bar, and I was also able to see the bar blink, maybe just setting per window will address both issue :)

Reply:
fixed